src/AutoBalance.cpp

```diff
@@ -407,7 +407,19 @@ class AutoBalance_AllMapScript : public AllMapScript
 
             AutoBalanceMapInfo *mapABInfo=map->CustomData.GetDefault<AutoBalanceMapInfo>("AutoBalanceMapInfo");
 
-            mapABInfo->playerCount--;// (maybe we've to found a safe solution to avoid player recount each time)
+            if ((map->GetEntry()->IsDungeon() || map->GetEntry()->IsRaid()) && player) {
+              Map::PlayerList const& playerList = map->GetPlayers();
+              for (Map::PlayerList::const_iterator playerIteration = playerList.begin(); playerIteration != playerList.end(); ++playerIteration) {
```

Comment:
this for cycle contains a return that will be always executed, it is so necessary to create a for cycle?
that `return` is right?

Reply:
the return is required, otherwise it will still scale down when a player leaves, the return; stops the scaling when a player left instance when group is in combat.